app/src/main/java/io/mochahub/powermeter/workoutsession/workoutsessiondialog/NewWorkoutViewModel.kt

```diff
@@ -0,0 +1,123 @@
+package io.mochahub.powermeter.workoutsession.workoutsessiondialog
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.room.withTransaction
+import io.mochahub.powermeter.data.AppDatabase
+import io.mochahub.powermeter.data.ExerciseEntity
+import io.mochahub.powermeter.data.WorkoutEntity
+import io.mochahub.powermeter.data.WorkoutSessionEntity
+import io.mochahub.powermeter.data.WorkoutSetEntity
+import io.mochahub.powermeter.models.Exercise
+import io.mochahub.powermeter.models.Workout
+import io.mochahub.powermeter.models.WorkoutSession
+import io.mochahub.powermeter.models.WorkoutSet
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+import java.time.Instant
+import java.util.Locale
+
+class NewWorkoutViewModel(val db: AppDatabase) : ViewModel() {
+    var workouts = ArrayList<Workout>()
```

Comment:
I think this should be a `LiveData` if so we can persist it on config changes without needing to override the app's config changes

Reply:
Turns out I didn't need the config changes. Removed the manifest and dialog code for config changes here https://github.com/power-meter/android-client/pull/32/commits/67e4c8c935c5f1e1c08d38f71487f2132c61b037 .

Reply:
If it is LiveData would it emit an onChange event on rotation?

Reply:
Yup it does